marquez_airflow/extractors/__init__.py

```diff
@@ -103,13 +118,27 @@ def get_operator_class(cls):
 
     def validate(self):
         # TODO: maybe we should also enforce the module
-        assert(self.operator_class is not None and
-               self.operator.__class__ == self.operator_class)
+        assert (self.operator_class is not None and
+                self.operator.__class__ == self.operator_class)
 
     def extract(self) -> [StepMetadata]:
-        raise NotImplementedError
+        step_metadata = self._extract()[0]
+
+        #
+        airflow_version = get_airflow_version()
+        if airflow_version:
+            step_metadata.context['airflow.version'] = airflow_version
+
+        #
+        step_metadata.context['marquez_airflow.version'] = VERSION
+
+        return [step_metadata]
+
+    @abstractmethod
+    def _extract(self) -> [StepMetadata]:
+        pass
```

Comment:
I think you should do the opposite. I mean that you should keep `extract` as the contract that the extractors have to implement (for backward compatibility) and introduce a new method (possibly `base_extract`) that is called by marquez_dag that introduces the extra behavior. By the way, that new method does not have to be in base extractor (but it can).

Reply:
You raise a good point, @julienledem. I think we'll eventually want to revisit the **BaseExtractor** contract, but introducing a breaking change now, as you mentioned, isn't ideal. I've added [`Utils.add_airflow_info_to(StepMetadata)`](https://github.com/MarquezProject/marquez-airflow/blob/feature/collect-airflow-version/marquez_airflow/utils.py#L127) to append common keys with the prefix `airflow.*`. We'll possibly even want to introduce a `JobContext` class to instantiate a context with common keys.